LunrCoreLmdb/KeyBuilder.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Text;
+
+namespace LunrCoreLmdb
+{
+    internal static class KeyBuilder
+    {
+        #region Fields
+
+        private static readonly byte[] FieldPrefix = Encoding.UTF8.GetBytes("F:");
+        public static byte[] BuildAllFieldsKey() => FieldPrefix;
+        public static byte[] BuildFieldKey(string field) => FieldPrefix.Concat(Encoding.UTF8.GetBytes(field));
+
+        #endregion
+
+        #region Field Vectors
+
+        private static readonly byte[] FieldVectorKeyPrefix = Encoding.UTF8.GetBytes("VK:");
+        public static byte[] BuildAllFieldVectorKeys() => FieldVectorKeyPrefix;
+        public static ReadOnlySpan<byte> BuildFieldVectorKeyKey(string key) => FieldVectorKeyPrefix.Concat(Encoding.UTF8.GetBytes(key));
+
+        private static readonly byte[] FieldVectorValuePrefix = Encoding.UTF8.GetBytes("V:");
+        public static byte[] BuildFieldVectorKey(string key) => FieldVectorValuePrefix.Concat(Encoding.UTF8.GetBytes(key));
+
+        #endregion
+
+        #region Inverted Indices
+        
+        private static readonly byte[] InvertedIndexEntryPrefix = Encoding.UTF8.GetBytes("E:");
+
+        public static byte[] BuildInvertedIndexEntryKey(string key) => InvertedIndexEntryPrefix.Concat(Encoding.UTF8.GetBytes(key));
+
+        #endregion
+
+        #region TokenSet
+
+        private static readonly byte[] TokenSetWordPrefix = Encoding.UTF8.GetBytes("T:");
+        public static byte[] BuildTokenSetWordKey(string word) => TokenSetWordPrefix.Concat(Encoding.UTF8.GetBytes(word));
+        public static byte[] BuildAllTokenSetWordKeys() => TokenSetWordPrefix;
+
+        #endregion
+
+        public static byte[] Concat(this byte[] left, byte[] right)
```

Comment:
Curious how much better than a simple `left.Concat(right).ToArray()` this is.

Reply:
It's a factor of 10 in favor of BlockCopy.

``` ini

BenchmarkDotNet=v0.12.1, OS=Windows 10.0.18363.1256 (1909/November2018Update/19H2)
Intel Core i9-9820X CPU 3.30GHz, 1 CPU, 20 logical and 10 physical cores
.NET Core SDK=5.0.101
  [Host]     : .NET Core 3.1.10 (CoreCLR 4.700.20.51601, CoreFX 4.700.20.51901), X64 RyuJIT
  Job-LSRXXT : .NET Core 3.1.10 (CoreCLR 4.700.20.51601, CoreFX 4.700.20.51901), X64 RyuJIT

RunStrategy=Throughput  

```
|     Method |      Mean |    Error |   StdDev |
|----------- |----------:|---------:|---------:|
|  BlockCopy |  14.71 ns | 0.052 ns | 0.049 ns |
| LinqConcat | 145.02 ns | 0.374 ns | 0.332 ns |

Reply:
Wow, amazing!